### DISCUSSION TITLE
Random Input Stuttering

### DISCUSSION THREAD

**Discussion by RuinTD** [ppy/osu#24112] [Q&A]
### Type

Performance

### Bug description

Hello. I've been encountering an issue with Lazer ever since upgrading to Windows 11 when it released in 2021 where my input occasionally freezes for a split second. I'm only now reporting this issue because I had always assumed it was a known issue with Windows 11 itself and with high polling rate mice, however [that seems to be fixed now](https://blogs.windows.com/windowsdeveloper/2023/05/26/delivering-delightful-performance-for-more-than-one-billion-users-worldwide/#:~:text=Reduced%20game%20stutter%20with%20high%20report%20rate%20mice).

This issue only occurs on Windows 11 and only on Lazer. It does not occur on osu! stable or in any other game.

### Screenshots or videos

https://www.youtube.com/watch?v=TsSdzCpWa1I

### Version

2023.621.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/11940309/database.log)
[input.log](https://github.com/ppy/osu/files/11940310/input.log)
[network.log](https://github.com/ppy/osu/files/11940311/network.log)
[performance.log](https://github.com/ppy/osu/files/11940312/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/11940313/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/11940314/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/11940315/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/11940316/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/11940317/runtime.log)
[updater.log](https://github.com/ppy/osu/files/11940318/updater.log)

**Comment by bdach** [comment]
Yeah well the very long input thread frame times will be the cause of that.

If you have any custom device drivers installed then please try disabling or removing them first. Also would be good to know *what* input devices you have plugged in on your system.

**Reply by RuinTD** [reply]
> If you have any custom device drivers installed

All my drivers come from Windows Update if that's what you mean?
Only input devices I have are [my mouse](https://www.amazon.com/dp/B09CXQZMHG), [my keyboard](https://www.amazon.com/dp/B07D5TNTPC) (both at 250 Hz/4 ms Polling Rate). and a Stream Deck, if that counts.

**Reply by bdach** [reply]
Probably the way to troubleshoot is to try and unplug each device one by one and see if the situation improves to identify which one may be at fault.

**Comment by peppy** [comment]
The input time overhead is showing external processes affecting this (grey on the frame time graph implies this is a delay we have no control over). Basically your assumption is correct, and something else on your system is hooking your mouse.

Solution: 
- reduce polling rate
- close / delete apps which are hooking the mouse using global input hooks

Why only lazer? Probably because other games are using hardware cursors and windows optimises for that.

**Reply by RuinTD** [reply]
Hey, peppy, thanks for the help! Do you know of any applications that could tell me what apps are hooking the mouse? I don’t think any of my apps would be doing that.

**Reply by peppy** [reply]
Anything / everything running on your system. Shut it down. One by one. Including things in system tray / "drivers" / anything.

**Reply by peppy** [reply]
https://osu.ppy.sh/help/wiki/Performance_Troubleshooting
